homeassistant/components/mqtt/__init__.py

```diff
@@ -774,6 +800,19 @@ async def finish_dump(_):
         ),
     )
 
+    # setup platforms and discovery
+    hass.data[DATA_CONFIG_ENTRY_LOCK] = asyncio.Lock()
```

Comment:
Sure, but the for now this change would be too big since `hass.data[DATA_MQTT]` is now the MQTT client.
And `DATA_MQTT` == `mqtt`. But sure something for a future PR.